### ISSUE TITLE
Trouble with getting device expiry events

### ISSUE THREAD

**Issue by DocSunset** [libmapper/libmapper#67]
I am having trouble getting device expiry events to show up at all in my python script. See the working example below. After starting the script, launch another device from another script. The device callback gets called as expected with `Event.NEW` and then `Event.MODIFIED`. I then halt the other device. No matter how long I wait, the device callback never seems to get called again. I was expecting it to eventually be called with `Event.REMOVED` or `Event.EXPIRED`.

I have tried to copy from Webmapper, but I guess I must be missing something?

Any help would be appreciated.

```py
import sys
from PySide6 import QtCore, QtWidgets
import libmapper as mpr

class MainWidget(QtWidgets.QWidget):
  def __init__(self):
    super().__init__()
    self.dev = mpr.Device("MishMash")
    self.dev.graph().add_callback(types=mpr.Type.DEVICE
        , func=lambda t, dev, evt: self.device_callback(dev, evt)
        )
    self.dev.graph().subscribe(None, mpr.Type.OBJECT, -1)
    self.libmapper_timer = QtCore.QTimer()
    self.libmapper_timer.timeout.connect(self.libmapper_poll)
    self.libmapper_timer.start(10)

  @QtCore.Slot()
  def libmapper_poll(self):
    self.dev.poll(0)

  def device_callback(self, dev, evt):
    if evt == mpr.Graph.Event.NEW:
      print("{} new".format(dev))
      self.dev.graph().subscribe(dev, mpr.Type.OBJECT, -1)
    elif evt == mpr.Graph.Event.MODIFIED:
      print("{} modified".format(dev))
    elif evt == mpr.Graph.Event.REMOVED:
      print("{} removed".format(dev))
    elif evt == mpr.Graph.Event.EXPIRED:
      print("{} expired".format(dev))

if __name__ == "__main__":
  app = QtWidgets.QApplication([])
  widget = MainWidget()
  widget.show()
  sys.exit(app.exec())
```

**Comment by DocSunset** [comment]
Here's a c equivalent that also doesn't seem to get expiry events. Presumably I am just missing something?

This is somewhat less relevant for my immediate project since #66 was closed.

```c
#include <signal.h>
#include <stdio.h>
#include <mapper/mapper.h>

int done = 0;

void sigint_handler(int sig)
{
    done = 1;
}

void print_evt_type(mpr_graph_evt evt)
{
    switch (evt)
    {
        case MPR_OBJ_NEW:
            printf("new ");
            break;
        case MPR_OBJ_MOD:
            printf("modified ");
            break;
        case MPR_OBJ_REM:
            printf("removed ");
            break;
        case MPR_OBJ_EXP:
            printf("new ");
            break;
    }
}

void print_obj_type(mpr_obj obj)
{
    mpr_type t = mpr_obj_get_type(obj); 
    if (t & MPR_DEV)     printf("device ");
    if (t & MPR_SIG_IN)  printf("destination ");
    if (t & MPR_SIG_OUT) printf("source ");
    if (t & MPR_SIG)     printf("signal ");
    if (t & MPR_MAP_IN)  printf("incoming ");
    if (t & MPR_MAP_OUT) printf("outgoing ");
    if (t & MPR_MAP)     printf("map ");
}

void graph_handler(mpr_graph gph, mpr_obj obj, const mpr_graph_evt evt, const void *data)
{
    printf("graph: ");
    print_evt_type(evt);
    print_obj_type(obj);
    printf("\n");
    if (evt == MPR_OBJ_NEW && mpr_obj_get_type(obj) & MPR_DEV)
        mpr_graph_subscribe(gph, obj, MPR_OBJ, -1);
}

int main()
{
    mpr_dev dev = mpr_dev_new("Devon", 0);
    mpr_graph gph = mpr_obj_get_graph(dev);
    signal(SIGINT, sigint_handler);
    mpr_graph_add_cb(gph, graph_handler, MPR_OBJ, 0);
    mpr_graph_subscribe(gph, 0, MPR_OBJ, -1);
    printf("start\n");
    while (done == 0) mpr_dev_poll(dev, 100);
    printf("done\n");
    return 0;
}
```

**Comment by malloch** [comment]
Hi @DocSunset thanks for the bug report! The handling of expired devices was only being handled when the graph was polled explicitly using `mpr_graph_poll()`. Should be fixed now in 517d102.
